OrbitGl/ProcessesDataView.cpp

```diff
@@ -256,38 +208,45 @@ void ProcessesDataView::DoFilter() {
 
 //-----------------------------------------------------------------------------
 void ProcessesDataView::UpdateProcessList() {
-  size_t numProcesses = m_ProcessList.Size();
+  size_t numProcesses = process_list_.size();
   m_Indices.resize(numProcesses);
   for (size_t i = 0; i < numProcesses; ++i) {
     m_Indices[i] = i;
   }
 }
 
 //-----------------------------------------------------------------------------
-void ProcessesDataView::SetRemoteProcessList(ProcessList a_RemoteProcessList) {
-  m_IsRemote = true;
-  m_ProcessList = std::move(a_RemoteProcessList);
+void ProcessesDataView::SetProcessList(
+    const std::vector<std::shared_ptr<Process>>& process_list) {
+  process_list_ = process_list;
   UpdateProcessList();
   OnSort(m_SortingColumn, {});
   OnFilter(m_Filter);
   SetSelectedItem();
 }
 
 //-----------------------------------------------------------------------------
-void ProcessesDataView::SetRemoteProcess(
-    const std::shared_ptr<Process>& a_Process) {
-  std::shared_ptr<Process> targetProcess =
-      m_ProcessList.GetProcess(a_Process->GetID());
-  if (targetProcess) {
-    m_SelectedProcess = a_Process;
-    UpdateModuleDataView(m_SelectedProcess);
+void ProcessesDataView::UpdateProcess(const std::shared_ptr<Process>& process) {
+  auto it =
+      std::find_if(process_list_.begin(), process_list_.end(),
+                   [&process](const std::shared_ptr<Process>& target_process) {
+                     return target_process->GetID() == process->GetID();
+                   });
+
+  if (it != process_list_.end()) {
+    *it = process;
   } else {
-    m_RemoteProcess = a_Process;
+    ERROR(
+        "Unable to update process \"%s\" pid: %d, the process is not in the "
+        "process list.",
```

Comment:
I would write it `"Unable to update process \"%s\" with pid %d: the process is not in the process list."`

Reply:
done